src/app/file-tree/file-tree.component.ts

```diff
@@ -62,7 +62,11 @@ export class FileTreeComponent {
     this.treeControl = new FlatTreeControl(this.getLevel, this.isExpandable);
     this.dataSource = new MatTreeFlatDataSource(this.treeControl, this.treeFlattener);
     this.dataSource.data = this.convertSourceFilesToTree(data.files);
-    this.primaryDescriptorPath = '';
+    this.dataSource.data.forEach((fileNode) => {
```

Comment:
This will work, but you're looking for the one primary descriptor file, so something like a `find` instead of a loop that goes through all source files regardless seems more appropriate (untested):
```
this.primaryDescriptorPath = this.dataSource.data
   .find(fileNode => fileNode.absolutePath === primaryDescriptorPath)?.absolutePath || '';
```